Content.Shared/Changeling/Transform/SharedChangelingTransformSystem.cs

```diff
@@ -0,0 +1,238 @@
+
+        currentDna.DNA = targetConsumedDna.DNA;
+
+//        CopyComps(targetIdentity, ent, null,  (IComponent) typeof(VocalComponent), (IComponent) typeof(SpeechComponent));
+        CopyComp<VocalComponent>(targetIdentity, ent, out _);
+        CopyComp<SpeechComponent>(targetIdentity, ent, out _);
+        // Make sure the target Identity has a Typing indicator, if the identity is human or dwarf and never had a mind it'll never have a typingIndicatorComponent
+        EnsureComp<TypingIndicatorComponent>(targetIdentity, out var targetTypingIndicator);
+        CopyComp<TypingIndicatorComponent>(targetIdentity, ent, out _);
+
+        //TODO: While it would be splendid to be able to provide the original owning player who was playing the targetIdentity, it's not exactly feasible to do
+        _adminLogger.Add(LogType.Action, LogImpact.High, $"{ToPrettyString(ent.Owner):player}  successfully transformed into \"{Name(targetIdentity)}\"");
+        _metaSystem.SetEntityName(ent, Name(targetIdentity), raiseEvents: false);
+        _metaSystem.SetEntityDescription(ent, MetaData(targetIdentity).EntityDescription);
+        TransformGrammarSet(ent, targetConsumedHumanoid.Gender);
+
+        Dirty(ent);
+    }
+
+    protected virtual void TransformBodyEmotes(EntityUid uid, EntityUid target) { }
+    public virtual void TransformGrammarSet(EntityUid uid, Gender gender) { }
+    protected virtual void StartSound(Entity<ChangelingTransformComponent> ent, SoundSpecifier? sound) { }
+    protected virtual void StopSound(Entity<ChangelingTransformComponent> ent) { }
+}
+
+
+public sealed partial class ChangelingTransformActionEvent : InstantActionEvent
+{
+
+}
+[Serializable, NetSerializable]
+public sealed partial class ChangelingTransformWindupDoAfterEvent : SimpleDoAfterEvent
+{
+    public NetEntity TargetIdentity;
+
+    public ChangelingTransformWindupDoAfterEvent(NetEntity targetIdentity)
+    {
+        TargetIdentity = targetIdentity;
+    }
+}
+
+[Serializable, NetSerializable]
+public sealed class ChangelingTransformIdentitySelectMessage : BoundUserInterfaceMessage
+{
+    public readonly NetEntity TargetIdentity;
+
+    public ChangelingTransformIdentitySelectMessage(NetEntity targetIdentity)
+    {
+        TargetIdentity = targetIdentity;
+    }
+}
+
+[Serializable, NetSerializable]
+public sealed class ChangelingIdentityData
+{
+    public readonly NetEntity Identity;
+    public string Name;
+
+
+    public ChangelingIdentityData(NetEntity identity, string name)
+    {
+        Identity = identity;
+        Name = name;
+    }
+}
+[Serializable, NetSerializable]
+public sealed class ChangelingTransformBoundUserInterfaceState : BoundUserInterfaceState
+{
+    public readonly List<ChangelingIdentityData> Identites;
+    public ChangelingTransformBoundUserInterfaceState(List<ChangelingIdentityData> identities)
+    {
+        Identites = identities;
+    }
+}
+[Serializable, NetSerializable]
+public enum TransformUi : byte
+{
+    Key,
+}
```

Comment:
```suggestion
[Serializable, NetSerializable]
public sealed class ChangelingTransformIdentitySelectMessage : BoundUserInterfaceMessage
{
    public readonly NetEntity TargetIdentity;

    public ChangelingTransformIdentitySelectMessage(NetEntity targetIdentity)
    {
        TargetIdentity = targetIdentity;
    }
}

[Serializable, NetSerializable]
public sealed class ChangelingIdentityData
{
    public readonly NetEntity Identity;
    public string Name;

    public ChangelingIdentityData(NetEntity identity, string name)
    {
        Identity = identity;
        Name = name;
    }
}

[Serializable, NetSerializable]
public sealed class ChangelingTransformBoundUserInterfaceState : BoundUserInterfaceState
{
    public readonly List<ChangelingIdentityData> Identites;
    public ChangelingTransformBoundUserInterfaceState(List<ChangelingIdentityData> identities)
    {
        Identites = identities;
    }
}

[Serializable, NetSerializable]
public enum TransformUi : byte
{
    Key,
}
```
Github sucks at showing this diff so... whitespaces between stuffs